spyder/plugins/projects/widgets/main_widget.py

```diff
@@ -598,6 +601,142 @@ def show_widget(self):
         self.raise_()
         self.update()
 
+    def handle_switcher_modes(self, mode):
+        """
+        Populate switcher with files in active project.
+        List the file names of the current active project with their
+        directories in the switcher. Only handle file mode, where
+        `mode` is empty string.
+        Parameters
+        ----------
+        mode: str
+            The selected mode (open files "", symbol "@" or line ":").
+        """
+        if mode != '':
+            return
+
+        paths = self._execute_fzf_subprocess()
+        if paths == []:
+            return
+        # the paths that are opened in the editor need to be excluded because
+        # they are shown already in the switcher in the "editor" section.
+        open_files = self.get_plugin()._get_open_filenames()
+        for file in open_files:
+            if file in paths:
+                paths.remove(file)
+
+        is_unsaved = [False] * len(paths)
+        short_paths = shorten_paths(paths, is_unsaved)
+        section = self.get_title()
+
+        for i, (path, short_path) in enumerate(zip(paths, short_paths)):
+            title = osp.basename(path)
+            icon = get_file_icon(path)
+            description = osp.dirname(path)
+            if len(path) > 75:
+                description = short_path
+            is_last_item = (i+1 == len(paths))
+
+            self.get_plugin()._switcher.add_item(
+                title=title,
+                description=description,
+                icon=icon,
+                section=section,
+                data=path,
+                last_item=is_last_item
+            )
+        self.get_plugin()._switcher.set_current_row(0)
```

Comment:
Maybe this part of the code could go in the plugin side? So in the plugin's `handle_switcher_modes` you call this method to get the items information but create the actual items in plugin method? So from the plugin you end up with something like:

```python
def handle_switcher_modes(self, mode):
    """
    ...
    """
    items = self.get_widget().handle_switcher_modes("")
    for title, description, icon, section, data, last_item in items:
        self._switcher.add_item(
            title=title,
            description=description,
            icon=icon,
            section=section,
            data=path,
            last_item=is_last_item
        )
    self._switcher.set_current_row(0)
```

Reply:
I agree with @dalthviz suggestion.